sanic_openapi/spec.py

```diff
@@ -1,47 +1,89 @@
-from sanic import Sanic
+from typing import Dict
 
 
 class Spec:
-    def __init__(self, app: Sanic) -> None:
+    def __init__(self, config) -> None:
+        self.config = config
         self.swagger = "2.0"
+        self._tags = {}
+        self._paths = {}
+        self._definitions = {}
 
-        self.info = {
-            "version": getattr(app.config, "API_VERSION", "1.0.0"),
-            "title": getattr(app.config, "API_TITLE", "API"),
-            "description": getattr(app.config, "API_DESCRIPTION", ""),
-            "termsOfService": getattr(app.config, "API_TERMS_OF_SERVICE", ""),
-            "contact": {"email": getattr(app.config, "API_CONTACT_EMAIL", None)},
+    @property
+    def info(self):
+        return {
+            "version": getattr(self.config, "API_VERSION", "1.0.0"),
+            "title": getattr(self.config, "API_TITLE", "API"),
+            "description": getattr(self.config, "API_DESCRIPTION", ""),
+            "termsOfService": getattr(self.config, "API_TERMS_OF_SERVICE", ""),
+            "contact": {"email": getattr(self.config, "API_CONTACT_EMAIL", None)},
             "license": {
-                "name": getattr(app.config, "API_LICENSE_NAME", None),
-                "url": getattr(app.config, "API_LICENSE_URL", None),
+                "name": getattr(self.config, "API_LICENSE_NAME", None),
+                "url": getattr(self.config, "API_LICENSE_URL", None),
             },
         }
-        self.schemes = getattr(app.config, "API_SCHEMES", ["http"])
 
-        self.host = getattr(app.config, "API_HOST", None)
+    @property
+    def schemes(self):
+        return getattr(self.config, "API_SCHEMES", ["http"])
+
+    @property
+    def host(self):
+        return getattr(self.config, "API_HOST", None)
+
+    @property
+    def basePath(self):
+        return getattr(self.config, "API_BASEPATH", None)
 
-        self.basePath = getattr(app.config, "API_BASEPATH", None)
+    # --------------------------------------------------------------- #
+    # Authorization
+    # --------------------------------------------------------------- #
 
-        # --------------------------------------------------------------- #
-        # Authorization
-        # --------------------------------------------------------------- #
+    @property
+    def securityDefinitions(self):
+        return getattr(self.config, "API_SECURITY_DEFINITIONS", None)
 
-        self.securityDefinitions = getattr(app.config, "API_SECURITY_DEFINITIONS", None)
-        self.security = getattr(app.config, "API_SECURITY", None)
+    @property
+    def security(self):
+        return getattr(self.config, "API_SECURITY", None)
+
+    @property
+    def definitions(self):
+        return self._definitions
+
+    @property
+    def tags(self):
+        return self._tags
+
+    @property
+    def paths(self):
+        return self._paths
 
         self.definitions = {}
         self.tags = []
         self.paths = {}
```

Comment:
I think this is because of the incorrect merge with `master` branch. 🤔

Reply:
Well spotted. Fixed